src/executables/util_cs.c

```diff
@@ -3296,17 +3328,41 @@ applyinfo (UTIL_FUNCTION_ARG * arg)
 	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
 				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
+	  if (error != NO_ERROR)
+	    {
+	      goto print_applyinfo_usage;
+	    }
 	}
 
     check_applied_info_end:
       if (error != NO_ERROR)
 	{
 	  PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+	  goto print_applyinfo_usage;
 	}
       error = NO_ERROR;
 
       if (check_master_info)
 	{
+	  check_master_info = false;
```

Comment:
`check_master_info` is a flag to determine whether to check or not, and you're using it in another way for error checking. I recommend using another variable for another purpose.

Reply:
Right. I will consider flag(0,1) or other variable.